src/jit/lsraarmarch.cpp

```diff
@@ -738,8 +738,22 @@ void Lowering::TreeNodeInfoInitPutArgStk(GenTreePutArgStk* argNode, fgArgTabEntr
     else
     {
 #ifdef _TARGET_ARM_
+
+#ifndef ARM_SOFTFP
         // We must not have a multi-reg struct; double uses 2 slots and isn't a multi-reg struct
         assert((info->numSlots == 1) || ((info->numSlots == 2) && (putArgChild->TypeGet() == TYP_DOUBLE)));
+#else  // ARM_SOFTFP
+        // The `double` types have been transformed to `long` on armel.
+        if ((info->numSlots == 2) && (putArgChild->TypeGet() == TYP_LONG))
+        {
+            argNode->gtLsraInfo.srcCount = 2;
+        }
+        else
+        {
+            assert(info->numSlots == 1);
+        }
+#endif // ARM_SOFTFP
+
```

Comment:
> updating srcCount was needed for ARM_SOFTFP

I see; in that case, it's probably better to set `isDouble` conditionally instead, and move the srcCount update into the same `#ifdef`. How about this:
```
#ifdef _TARGET_ARM_

#ifdef ARM_SOFTFP
        // The `double` types have been transformed to `long` on armel.
        const bool isDouble = (info->numSlots == 2) && (putArgChild->TypeGet() == TYP_LONG);
        if (isDouble)
        {
            argNode->gtLsraInfo.srcCount = 2;
        }
#else
        const bool isDouble = (info->numSlots == 2) && (putArgChild->TypeGet() == TYP_DOUBLE);
#endif

        // We must not have a multi-reg struct; double uses 2 slots and isn't a multi-reg struct
        assert((info->numSlots == 1) || isDouble);

#else  // !_TARGET_ARM_
        // We must not have a multi-reg struct
        assert(info->numSlots == 1);
#endif // !_TARGET_ARM_
```